astroquery/xmm_newton/core.py

```diff
@@ -133,8 +134,16 @@ def download_data(self, observation_id, filename=None, verbose=False,
         link = link + "".join("&{0}={1}".format(key, val)
                               for key, val in kwargs.items())
 
-        response = self._request('GET', link, save=True)
-        if response is not None:
+        if verbose:
+            log.info(link)
+
+        response = None
+        try:
+            response = self._request('GET', link, save=True, cache=True)
+        except HTTPError:
+            log.error("Unable to access URL {0}".format(link))
+
+        if response:
```

Comment:
This shouldn't be needed, nor should line 140 above.  A `response` cannot be `None`